src/cljs/athens/main/core.cljs

```diff
@@ -26,14 +26,49 @@
     (.. ^js @main-window -webContents (send text))))
 
 
+(defn init-electron-handlers
+  []
+  (let [toggle-max-or-min-win-channel "toggle-max-or-min-active-win"
+        close-win-channel "close-win"
+        exit-fullscreen-win-channel "exit-fullscreen-win"]
+    (doto ipcMain
+      (.handle toggle-max-or-min-win-channel
+               (fn [_ toggle-min?]
+                 (when-let [active-win (.getFocusedWindow BrowserWindow)]
+                   (if toggle-min?
+                     (if (.isMinimized active-win)
+                       (.restore active-win)
+                       (.minimize active-win))
+                     (if (.isMaximized active-win)
+                       (.unmaximize active-win)
+                       (.maximize active-win))))))
+      (.handle close-win-channel
+               (fn []
+
+                 (.quit app)))
+      (.handle exit-fullscreen-win-channel
+               (fn []
+                 (when-let [active-win (.getFocusedWindow BrowserWindow)]
+                   (.setFullScreen active-win false)))))
+
+    ;; Future intent to refactor statup to use startup and teardown effects
+    ;; Below is an example of the teardown effect for this init fn
+    ;; #(doall (.removeHandler ipcMain toggle-max-or-min-win-channel)
+    ;;         (.removeHandler ipcMain close-win-channel)
+    ;;         (.removeHandler ipcMain exit-fullscreen-win-channel))
```

Comment:
Why future?

Reply:
There were a lot of changes we weren't sure about. We went on the side of keeping scope narrow until you could review the toolbar specific changes. We can work on including this refactor.